src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -128,12 +131,81 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
+        }
+
+        internal static async Task<Range?> TryGetPropertyRangeAsync(RazorCodeDocument codeDocument, string propertyName, RazorDocumentMappingService documentMappingService, CancellationToken cancellationToken)
+        {
+            // Parse the C# file and find the property that matches the name, and that has a [Parameter] attribute
+
+            var csharpText = codeDocument.GetCSharpSourceText();
+            var syntaxTree = CSharpSyntaxTree.ParseText(csharpText, cancellationToken: cancellationToken);
+            var root = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var property = root.DescendantNodes(n => n is BaseNamespaceDeclarationSyntax or CompilationUnitSyntax or ClassDeclarationSyntax)
+                .OfType<PropertyDeclarationSyntax>()
+                .Where(p => p.Identifier.ValueText.Equals(propertyName, StringComparison.Ordinal))
+                .FirstOrDefault();
+
+            // Did we find a property at all?
+            if (property is null)
+            {
+                return null;
+            }
+
+            // We might have found a property, but is it a parameter?
+            foreach (var attributeList in property.AttributeLists)
+            {
+                foreach (var attribute in attributeList.Attributes)
+                {
+                    // Attributes could be simple like [Parameter] or qualified like [Blah.Parameter]
+                    // but we don't care about that distinction, so lets just avoid dealing with the complexities
+                    // of the roslyn tree.
+                    var name = attribute.ToString();
+
+                    // Sadly we don't have symbolic information here, so we have to just hope for the best
+                    // Since we're only navigating its not a big deal, plus its not possible to have multiple
+                    // properties with the same name defined, so its not like one could be a real parameter, and
+                    // another could be a fake one
+                    if (name.Equals("ParameterAttribute", StringComparison.Ordinal) ||
+                        name.Equals("Parameter", StringComparison.Ordinal) ||
+                        name.EndsWith(".ParameterAttribute", StringComparison.Ordinal) ||
+                        name.EndsWith(".Parameter", StringComparison.Ordinal))
+                    {
+                        var range = property.Identifier.Span.AsRange(csharpText);
+                        if (documentMappingService.TryMapFromProjectedDocumentRange(codeDocument, range, out var originalRange))
+                        {
+                            return originalRange;
+                        }
+
+                        // if we found the property we wanted, but couldn't map, there is no point doing any more work
+                        return null;
+                    }
+                }
+            }
+
+            return null;
```

Comment:
So this scenario would be like someone tried to navigate to a property, and it exists on the class but doesn't have the proper attributes applied? It seems like that maybe shouldn't even be possible in real use-cases, should we maybe log here?

Reply:
Yes, and yes that shouldn't be possible. See my comment in the other thread, I think I'll get rid of the attribute checking entirely. Pit of success, and all that.